robot_smach_states/src/robot_smach_states/navigation/constraint_functions/symbolic_constraints.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import absolute_import
+
+# ROS
+import rospy
+
+
+def symbolic_constraint(robot, entity_designator_area_name_map):
+    """
+    Generate the constraint based on an entity and an area of that entity.
+    :param robot: robot object
+    :param entity_designator_area_name_map: dictionary mapping EdEntityDesignators to a string or designator
+        resolving to a string, representing the area, e.g., entity_designator_area_name_map[<EdEntity>] = 'in_front_of'.
+    :return: (tuple(PositionConstraint, OrientationConstraint)). If one of the entities does not resolve,
+        None is returned.
+    """
+    entity_id_area_name_map = {}
+    for desig, area_name in entity_designator_area_name_map.iteritems():
+        entity = desig.resolve()
+        try:
+            area_name = area_name.resolve()
+        except:
+            pass
```

Comment:
This is an artefact from the previous code. It was meant to make it so that you could either use a designator or a value (which is a bad practice but not the focus of this PR). I made it less ugly by checking for a "resolve_type" attribute